File: DAY 8/solutions.py/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map={}
        for i in range(len(nums)):
            required=target-nums[i]
            if required in map:
                return[map[required],i]
            map[nums[i]]=i
        return[-1,-1]        
        

File: DAY 8/solutions.java/1. Two Sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer, Integer> hashmap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {

            int needed = target - nums[i];

            if (hashmap.containsKey(needed)) {
                return new int[] {hashmap.get(needed), i};
            }

            hashmap.put(nums[i], i);
        }

        return new int[] {};
    }
}